subsys/storage/stream/stream_flash.c

```diff
@@ -181,6 +181,27 @@ size_t stream_flash_bytes_written(struct stream_flash_ctx *ctx)
 	return ctx->bytes_written;
 }
 
+struct _inspect_flash {
+	size_t buf_len;
+	size_t total_size;
+};
+
+static bool _stream_inspect_flash(const struct flash_pages_info *info,
+				  void *data)
+{
+	struct _inspect_flash *cb = (struct _inspect_flash *) data;
+
+	if (cb->buf_len > info->size) {
+		LOG_ERR("Buffer size is bigger than page");
+		cb->total_size = 0;
+		return false;
+	}
```

Comment:
This could probably be left for conditional compilation; the moment everything gets tested this should never happen.

Reply:
Seems that the function will not be called too often so there is, probably, no reason to bother with turning this off.

Reply:
Above `if` check compatibility stream_flash configuration whit given flash_area. The check is trivial (and no run-time costly) and it is sane in development phase.

Reply:
> 
> 
> Above `if` check compatibility stream_flash configuration whit given flash_area. The check is trivial (and no run-time costly) and it is sane in development phase.

Yes, I have got to the same conclusion.

There should be some method to figure out flash size at compilation, unless it is one that can be hot-plugged or is undetermined at compilation, because in most cases that would be simple number that can be provided via dts, instead we are calculating it at run-time. Just an observation.